rootfs/etc/nginx/lua/balancer/sticky.lua

```diff
@@ -73,7 +74,17 @@ function _M.balance(self)
   if not key then
     local random_str = string.format("%s.%s", ngx.now(), ngx.worker.pid())
     key = encrypted_endpoint_string(self, random_str)
-    set_cookie(self, key)
+  end
+
+  if not self.cookie_locations then
+    return self.instance:find(key)
+  end
+
+  for _, path in ipairs(self.cookie_locations[ngx.var.host]) do
+    if ngx.var.location_path == path then
+      set_cookie(self, key)
+      break
+    end
```

Comment:
I think it will be OK once you fix https://github.com/kubernetes/ingress-nginx/pull/3383/files#r233362108 since for those locations there won't be a cookie and therefore key will be generated randomly, which will make sure requests to those locations don't always go to the same upstream peer.

Reply:
It will generate a new key if the cookie is not found though, so it will be choose a peer based on chash. Defaulting to round robin would require a separate balancer which doesn't seem right. I think that some documentation is needed here, I'll add something.

Reply:
See: https://github.com/kubernetes/ingress-nginx/pull/3383/commits/3ceb2c28abc635102fbf22b66da96406c3dac344